### PR TITLE
[fix] update ```BasicStatistics``` and ```IncrementalBasicStatistics``` to follow additional sklearn conventions

### PR DESCRIPTION
**Pull request by icfaust**
### Description

Result attributes in sklearn have [trailing attributes](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes), ```BasicStatistics``` and ```IncrementalBasicStatistics``` follow this convention.  This converts all results to have trailing underscores, but still enables the original attributes with a warning. This adds a deprecation test and the warning tells of an API change in 2026.0.

It includes a change from ```options``` to ```result_options``` to allow for sklearn cloning in ```BasicStatistics```.

Additionally, it solves a number of missing aspects which would be covered if it were getting conformance tested by sklearn:

Adds the parameter check to ```BasicStatistics```.
Moves the parameter check before fit (where it is useful) in ```IncrementalBasicStatistics```.
Adds ```n_features_in_``` after fit for ```BasicStatistics```.
Fixes some docstring issues.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by olegkkruglov** [comment]
does it make sense to change IncrementalBasicStatistics in the same way here?

**Review by olegkkruglov — APPROVED** [review]
LGTM, thanks for the update

**Comment by samir-nasibli** [comment]
> does it make sense to change IncrementalBasicStatistics in the same way here?

I hope this is covered by ticket. @icfaust

**Review by samir-nasibli — APPROVED** [review]
Nice work! Good to go. Thank you @icfaust

**Review by Alexsandruss — APPROVED** [review]
